test/unit/object/test_task.py

```diff
@@ -0,0 +1,113 @@
+
+
+def test_get(test_task, mock_box_session):
+    expected_url = '{0}/tasks/{1}'.format(API.BASE_API_URL, test_task.object_id)
+    mock_box_session.get.return_value.json.return_value = {
+        'type': test_task.object_type,
+        'id': test_task.object_id,
+        'due_at': '2014-04-03T11:09:43-07:00',
+        'action': 'review',
+        'message': 'Test Message',
+    }
+    retrieved_task = test_task.get()
+    mock_box_session.get.assert_called_once_with(expected_url, headers=None, params=None)
+    assert isinstance(retrieved_task, Task)
+    assert retrieved_task.object_type == test_task.object_type
+    assert retrieved_task.object_id == test_task.object_id
+
+
+def test_update(test_task, mock_box_session):
+    new_message = 'New Message'
+    expected_url = '{0}/tasks/{1}'.format(API.BASE_API_URL, test_task.object_id)
+    mock_box_session.put.return_value.json.return_value = {
+        'type': test_task.object_type,
+        'id': test_task.object_id,
+        'message': new_message,
+    }
+    expected_body = {
+        'message': new_message,
+    }
+    updated_task = test_task.update_info({'message': new_message})
+    mock_box_session.put.assert_called_once_with(expected_url, data=json.dumps(expected_body), headers=None, params=None)
+    assert isinstance(updated_task, Task)
+    assert updated_task.message == new_message
+    assert updated_task.object_type == test_task.object_type
+    assert updated_task.object_id == test_task.object_id
+
+
+def test_delete_policy_return_the_correct_response(
+        test_task,
+        mock_box_session,
+        delete_task_response,
+):
+    # pylint:disable=redefined-outer-name
+    mock_box_session.delete.return_value = delete_task_response
+    response = test_task.delete()
+    # pylint:disable=protected-access
+    expected_url = test_task.get_url()
+    # pylint:enable = protected-access
+    mock_box_session.delete.assert_called_once_with(expected_url, params={}, expect_json_response=False, headers=None)
+    assert response is True
+
+
+def test_assign(test_task, mock_user, mock_box_session):
+    expected_url = '{0}/task_assignments'.format(API.BASE_API_URL)
+    expected_body = {
+        'task': {
+            'type': test_task.object_type,
+            'id': test_task.object_id,
+        },
+        'assign_to': {
+            'id': mock_user.object_id,
+            'login': 'test_user@example.com',
+        },
+    }
+    mock_box_session.post.return_value.json.return_value = {
+        'type': 'task_assignment',
+        'id': '42',
+    }
+    new_legal_hold_assignment = test_task.assign(assignee=mock_user, assign_to_login='test_user@example.com')
+    mock_box_session.post.assert_called_once_with(expected_url, data=json.dumps(expected_body))
+    assert isinstance(new_legal_hold_assignment, TaskAssignment)
+    assert new_legal_hold_assignment.object_type == 'task_assignment'
+    assert new_legal_hold_assignment.object_id == '42'
```

Comment:
Two things: these variable names need to be updated so they don't refer to legal holds, and you need to assert about the rest of the body to verify that the object translation worked correctly